gradio/blocks.py

```diff
@@ -974,6 +976,9 @@ def __init__(
         fill_height: bool = False,
         fill_width: bool = False,
         delete_cache: tuple[int, int] | None = None,
+        spa_mode: bool | None = None,
```

Comment:
is spa the best term to use? wouldn't "csr" be more appropriate here in contrast with ssr? Particularly after we add support for multiple pages

Reply:
Alternatively, we could just rename to:
```suggestion
        ssr_mode: bool | None = None,
```
if we go down this route, we'll need to rename the env variables, etc.

Reply:
The reason I went for SPA is because I _think_ spa is a more commonly understood term, and technically ssr mode is actually server side rendering with client side takeover, ssr is just one half of the process. 

Some of these terms are very jargony but spa is the one that most common for all software engineers. CSR isn't even that widely known amongst front end devs.

Reply:
Sg, reviewing the rest of the PR shortly!

Reply:
I'll fix those lint and type error